### PR TITLE
Added action to enable GitHub Pages on a repo

### PR DESCRIPTION
**Pull request by awanlin**
## Hey, I just made a Pull Request!

Added action to enable GitHub Pages on a repo

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder-backend-module-github/src/actions/githubPages.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  const { integrations, githubCredentialsProvider } = options;
+
+  return createTemplateAction<{
+    repoUrl: string;
+    buildType?: 'legacy' | 'workflow';
+    sourceBranch?: string;
+    sourcePath?: '/' | '/docs';
+    token?: string;
+  }>({
+    id: 'github:pages',
```
Comment (benjdlambert) [inline plugins/scaffolder-backend-module-github/src/actions/githubPages.ts:48]: should this be `github:pages:enable` or something instead?

Comment (awanlin) [reply plugins/scaffolder-backend-module-github/src/actions/githubPages.ts:48]: Sure, that crossed my mind when I was working on this. Should I update the various file names too? Feel like I should as there's some conventions for it.

Comment (benjdlambert) [reply plugins/scaffolder-backend-module-github/src/actions/githubPages.ts:48]: Yeah, thinking that `createGithubPagesEnableAction` etc makes more sense now if we're thinking of renaming.

**Review by benjdlambert — APPROVED** [review]
NIce! Thanks @awanlin!

**Comment by awanlin** [comment]
Thanks @benjdlambert

**Pull request merged**